[Backport v3.1-branch] doc: hpf and softperipherals comparison (#23658)
nrfconnect/sdk-nrf: doc/nrf/app_dev/device_guides/coprocessors/index.rst

   The High-Performance Framework (HPF) support in the |NCS| is :ref:`experimental <software_maturity>` and is limited to the nRF54L15 device.

You can use the VPR coprocessor (Fast Lightweight Peripheral Processor - FLPR) as follows:

* As an additional core in a multicore system using Zephyr in multithreaded mode (see the :ref:`nRF54L15<vpr_flpr_nrf54l>` and :ref:`nRF54H20 devices<ug_nrf54h20_flpr>` pages).
* As a peripheral emulator, using one of the following methods depending on the use case:

  * :ref:`High-Performance Framework (HPF)<hpf_index>`
  * :ref:`nrfxlib:soft_peripherals`

  .. note::

   In these usage modes, it is important to clearly differentiate between the Soft Peripheral and HPF solutions.
   Soft Peripherals serve as a direct replacement for hardware peripherals, offering guaranteed performance.
   In contrast, the HPF will allow you to accelerate protocol operations, but performance depends on your implementation.

>    We recommend using the Soft Peripheral solution if it meets your product's requirements.

Reviewer: ```suggestion
   It is recommended to use the Soft Peripheral solution if it meets your product's requirements.
```

Reviewer 2: I have added this suggestion in PR https://github.com/nrfconnect/sdk-nrf/pull/23631/files#diff-343961ffeca97c3468fdfc8e3638fabdbfbd980190d43a5d0b90dadc8a9c1938, so that this is fixed in both main and 3.1.0 branch. 
I think we can get this PR merged.
